app/controllers/api/v1/shops/books_controller.rb

```diff
@@ -0,0 +1,6 @@
+class Api::V1::Shops::BooksController < Api::V1::Shops::ApplicationController
+  def mark_as_sold
```

Comment:
Not a big deal too, but I would keep it restful, something like `Shops::BooksBatchController#update` since we're updating a list of resources

Reply:
Yes, it's a possible way. 
But almost everybody cook rest on their own, so I've decided to do like that.